staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go

```diff
@@ -52,11 +53,38 @@ type C struct {
 	I []interface{} `json:"ci"`
 }
 
-// C needs to implement runtime.Object to make it usable for tests.
+type D struct {
+	A []interface{} `json:"da"`
+}
+
+type E struct {
+	A interface{} `json:"ea"`
+}
+
+type F struct {
+	A string `json:"fa"`
```

Comment:
@smarterclayton - I just checked it, and it seems some of those doesn't have any chance to work.
Basically, if the input is incorrect, in some cases just doing:
1. string -> json.Unmarshal - > object
vs
2. string -> json.Unmarshal -> map[string]interface{} ->json.Marshal -> []byte -> json.Unmarshal -> object

Returns different results (i.e. the first one is error, the latter is not).

That said, since the "FromUnstructured" works on data from 2nd scenario, it doesn't have a way to error (so we cannot expect it to return the error).

Though, there are still some differences, which I will debug further next week.